src/main/mainWindow.js

```diff
@@ -63,34 +66,49 @@ function createMainWindow(config, options) {
   const indexURL = global.isDev ? 'http://localhost:8080/browser/index.html' : `file://${app.getAppPath()}/browser/index.html`;
   mainWindow.loadURL(indexURL);
 
-  // This section should be called after loadURL() #570
-  if (options.hideOnStartup) {
-    if (windowOptions.maximized) {
-      mainWindow.maximize();
-    }
-
-    // on MacOS, the window is already hidden until 'ready-to-show'
-    if (process.platform !== 'darwin') {
+  // handle hiding the app when launched by auto-start
+  if (hideOnStartup) {
+    if (trayIconShown && process.platform !== 'darwin') {
```

Comment:
No longer ignoring hideOnStartup when in macOS because this is also true when `--hidden` flag is passed via command line so we also need to minimize for macOS.